Source/Core/VideoCommon/AbstractFramebuffer.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+#include "Common/MathUtil.h"
+#include "VideoCommon/TextureConfig.h"
+
+class AbstractTexture;
+
+// An abstract framebuffer wraps a backend framebuffer/view object, which can be used to
+// draw onto a texture. Currently, only single-level textures are supported. Multi-layer
+// textures will render by default only to the first layer, however, multiple layers
+// be rendered in parallel using geometry shaders and layer variable.
+
+class AbstractFramebuffer
+{
+public:
+  AbstractFramebuffer(AbstractTextureFormat color_format, AbstractTextureFormat depth_format,
+                      u32 width, u32 height, u32 layers, u32 samples);
```

Comment:
What shall happen if the given dimension doesn't match the textures? If you are going to yield an error, why do you need to pass the size at all?

Or are you thinking about framebuffers without attachments?

Reply:
ValidateConfig checks the dimensions match, and the backends shouldn't create a framebuffer for an invalid configuration.

My motivation for storing a copy of the formats and dimensions in the framebuffer class was so we could have a second abstract framebuffer class to represent the backbuffer, where a corresponding abstract texture object does not exist. This way the callers can still access this information for setting viewports, etc.